pandas/io/excel/_util.py

```diff
@@ -21,16 +16,11 @@ def register_writer(klass):
         raise ValueError("Can only register callables as engines")
     engine_name = klass.engine
     _writers[engine_name] = klass
-    for ext in klass.supported_extensions:
```

Comment:
So just to be clear this was never part of the public API hence why I think OK to remove. If a user in spite of that still registered a custom subclass with an extension outside of `.xls`, `.xlsx` or `.xlsm` then potentially it would have an impact, but that seems like such an edge case to me hence why I think OK to remove.

cc @jreback in any case

Reply:
Hmm...I'm -0.5 on this.  Discarding because it's an "edge case" does not seem like a strong argument for removing this code.  Potentially there are better ways to implement this, but I wouldn't necessarily go as far as removing for now.

Reply:
I think the point is that it's not part of the public API and is a no-op with current internal code

Reply:
> My main point is that it's not part of the public API and is a no-op with current internal code

True.  I'm still a little on the fence on this though for the reason I gave earlier, so getting @jreback input would be helpful for this.

Reply:
this is ok i think